Source/WebKit/Shared/Cocoa/WKKeyedCoder.mm

```diff
@@ -148,4 +191,103 @@ - (NSDictionary *)accumulatedDictionary
     return m_dictionary.get();
 }
 
+- (void)encodeValueOfObjCType:(const char *)type at:(const void *)addr
+{
+    // NSURLRequest decoding calls to configure decode parameters
+
+    if (!m_NSURLVersionWrites && !strncmp(type, "i", 1)) {
+        int32_t NSURLRequestVersion = *(const int32_t *)addr;
+        [self encodeObject:[NSNumber numberWithInt: NSURLRequestVersion] forKey:@"wkcoder_nsurlrequest_version"];
+        m_NSURLVersionWrites = 1;
+        return;
+    }
+
+    if (m_NSURLVersionWrites == 1 && !strncmp(type, "i", 1)) {
+        int32_t cfVersion = *(const int32_t *)addr;
+        [self encodeObject:[NSNumber numberWithInt: cfVersion] forKey:@"wkcoder_nsurlrequest_cfversion"];
+        m_NSURLVersionWrites = 2;
+        return;
+    }
+
+    if (m_NSURLVersionWrites == 2 && !strncmp(type, "i", 1)) {
+        int32_t objectCount = *(const int32_t *)addr;
+        [self encodeObject:[NSNumber numberWithInt: objectCount] forKey:@"wkcoder_nsurlrequest_object_count"];
+        m_NSURLVersionWrites = 3;
+        return;
+    }
+
+    if (m_NSURLVersionWrites == 3)
+        ASSERT_NOT_REACHED_WITH_MESSAGE("WKKeyedCoder more attempts than expected calling -encodeValueOfObjCType:at:");
+}
+
+- (void)decodeValueOfObjCType:(const char *)type at:(void *)data size:(NSUInteger)size
+{
+    if (m_failedDecoding)
+        return;
+
+    // NSURLRequest -initWithCoder calls [coder allowedClasses] and then into decodeObjectOfClass* methods on WKKeyedCoder.
+    // This code is expecting to be called when decoding an NSURLRequest
+    m_handling_NSURLRequest = YES;
```

Comment:
This should be configured when making the WKKeyedCoder based on the need to decode an NSURLRequest.